File: app/src/main/java/lt/welovedotnot/ktu_ais_app/adapters/DrawerItemCustomAdapter.java
package lt.welovedotnot.ktu_ais_app.adapters;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import lt.welovedotnot.ktu_ais_app.R;

/**
 * Created by Mindaugas on 5/1/2017.
 */

public class DrawerItemCustomAdapter extends BaseAdapter {
    Context mContext;
    int layoutResourceId;
    String titles[] = null;

    public DrawerItemCustomAdapter(Context mContext, int layoutResourceId, String[] titles) {
        this.layoutResourceId = layoutResourceId;
        this.mContext = mContext;
        this.titles = titles;
    }

    @Override
    public int getCount() {
        return titles.length;
    }

    @Override
    public Object getItem(int i) {
        return null;
    }

    @Override
    public long getItemId(int i) {
        return 0;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = ((Activity) mContext).getLayoutInflater();
        View listItem = inflater.inflate(layoutResourceId, parent, false);

        TextView textViewName = (TextView) listItem.findViewById(R.id.drawerItemText);
        textViewName.setText(titles[position]);

        return listItem;
    }
}

File: app/src/main/java/lt/welovedotnot/ktu_ais_app/adapters/DrawerItemCustomAdapter.kt
package lt.welovedotnot.ktu_ais_app.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import lt.welovedotnot.ktu_ais_app.R
import lt.welovedotnot.ktu_ais_app.models.ScreenModel

/**
 * Created by Mindaugas on 5/1/2017.
 */

class DrawerItemCustomAdapter(context: Context, var titles: List<ScreenModel>) : BaseAdapter() {
    val inflater: LayoutInflater = LayoutInflater.from(context)

    override fun getCount(): Int {
        return titles.size
    }

    override fun getItem(i: Int): Any? {
        return null
    }

    override fun getItemId(i: Int): Long {
        return 0
    }

    override fun getView(position: Int, view: View?, parent: ViewGroup): View {
        var view = view
        if (view == null) {
            view = inflater.inflate(R.layout.drawer_list_item, parent, false)
        }

        val textViewName = view!!.findViewById(R.id.drawerItemText) as TextView
        textViewName.text = titles[position].name

        return view
    }
}
